ginga/canvas/types/basic.py

```diff
@@ -573,11 +573,15 @@ def select_contains(self, viewer, data_x, data_y):
         return self.select_contains_points(viewer, points, data_x, data_y)
 
     # TODO: this probably belongs somewhere else
-    def get_pixels_on_curve(self, image):
+    def get_pixels_on_curve(self, image, getvalues=True):
         data = image.get_data()
         wd, ht = image.get_size()
-        res = [ data[y, x] if 0 <= x < wd and 0 <= y < ht else numpy.NaN
-                for x, y in self.get_points_on_curve(image) ]
+        if getvalues:
+            res = [ data[y, x] if 0 <= x < wd and 0 <= y < ht else numpy.NaN
+                    for x, y in self.get_points_on_curve(image) ]
+        else:
+            res = [ [x, y] if 0 <= x < wd and 0 <= y < ht else numpy.NaN
+                    for x, y in self.get_points_on_curve(image) ]
```

Comment:
Hmm... not sure if you want `NaN`s mixed in with coordinate pairs.  This would cause basic numpy array transposition to fail.  Maybe we need a "magic" bad coordinate value.  At the very least this needs to return a pair instead of `NaN`.